apiserver/facades/client/client/status.go

```diff
@@ -1046,14 +1053,15 @@ func (c *statusContext) processUnit(ctx context.Context, unitName coreunit.Name,
 	// Handle any subordinate units.
 	result.Subordinates = make(map[string]params.UnitStatus)
 	for _, name := range subUnits {
-		subUnit, ok := c.unitByName(name)
+		subUnit, ok := c.units[name]
 		if !ok {
+			logger.Warningf(ctx, "missing subordinate unit %q", name)
 			continue
 		}
 
 		subCharmURL, ok := c.allAppsUnitsCharmBindings.applicationCharmURL[subUnit.ApplicationName]
 		if !ok {
-			logger.Debugf(ctx, "missing subordinate application %q", subUnit.ApplicationName)
+			logger.Warningf(ctx, "missing subordinate application %q", subUnit.ApplicationName)
```

Comment:
I don't think this needs to be warning. If this is on prodstack, this *could* be very chatty!

Reply:
I don't think so, I think it's only possible to execute this in the case of a programming error

Reply:
This would mean we have a unit which is subordinate to an application that doesn't exist

Reply:
Yeah, fair enough